src/analysis/normed_space/linear_isometry.lean

```diff
@@ -347,3 +348,38 @@ lemma linear_isometry.id_to_linear_map :
   (linear_isometry.id.to_linear_map : E →ₗ[R] E) = linear_map.id := rfl
 
 end linear_isometry_equiv
+
+namespace linear_isometry
+
+open finite_dimensional linear_map
+
+/-- A linear isometry between finite dimensional spaces of equal dimension can be upgraded
+    to a linear isometry equivalence. -/
+noncomputable def linear_isometry_equiv_of_finrank_eq_finrank {R X Y : Type*}
```

Comment:
I think we can shorten the name, maybe just `to_linear_isometry_equiv`, the condition should be clear to readers.